### PR TITLE
Updates Dependencies to latest

### PR DESCRIPTION
**Pull request by belcherj**
### Fix
Updates dependencies to latest

### Test
1. Run app, click around, look for errors

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated with:

- Updated Dependencies [#1848](https://github.com/Automattic/simplenote-electron/pull/1848)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,29 +88,29 @@
...
     "rimraf": "3.0.0",
-    "sass-loader": "8.0.0",
+    "sass-loader": "8.0.2",
     "spectron": "6.0.0",
-    "style-loader": "1.1.2",
-    "typescript": "3.7.4",
-    "wait-on": "3.3.0",
+    "style-loader": "1.1.3",
+    "typescript": "3.7.5",
+    "wait-on": "4.0.0",
```
Comment (dmsnell) [inline package.json]: do we even need this? this is a major update so I checked to see what we'd need to change and it's not referenced anyway. it came in #859 with `misty` and I'm not sure if it was meant to be a transitive/peer dependency. I haven't tried building without it

Comment (belcherj) [reply package.json]: https://github.com/Automattic/simplenote-electron/blob/develop/misty.yml#L6

Comment (belcherj) [reply package.json]: It is used.

Comment (belcherj) [reply package.json]: I don't think the API changed but they rewrote the code thus the full point bump.

**Pull request merged**